geom/geojson_unmarshal.go

```diff
@@ -263,8 +263,19 @@ func geojsonNodeToGeometry(node interface{}, ctype CoordinatesType, opts []Const
 		return poly.AsGeometry(), err
 	case geojsonMultiPoint:
 		// GeoJSON MultiPoints cannot contain empty Points.
-		seq := twoDimFloat64sToSequence(node.coords, ctype)
-		return NewMultiPoint(seq, opts...).AsGeometry(), nil
+		if len(node.coords) == 0 {
+			return MultiPoint{}.ForceCoordinatesType(ctype).AsGeometry(), nil
+		}
+		points := make([]Point, len(node.coords))
+		for i, coords := range node.coords {
+			coords, ok := oneDimFloat64sToCoordinates(coords, ctype)
+			if ok {
+				points[i] = NewPoint(coords, opts...)
+			} else {
+				points[i] = NewEmptyPoint(ctype)
+			}
```

Comment:
That's a pretty good idea. We could have something like:

```
// OptionalCoordinates represents Coordinates that may optionally be empty.
type OptionalCoordinates struct {
    // Coordinates hold the full coordinates when they are non-empty, or just
    // the CoordinateType when they are empty.
    Coordinates

    // Empty indicates if the XY, Z, and M parts of the Coordinates are empty
    // or populated.
    Empty bool
}
```

We could use this new type for the internals of `Point` as well. Since it will collapse together `NewPoint` and `NewEmptyPoint` (and there may be some fairly noisy flow on effects), I'll raise a new ticket for this and work on it before the NaN/Infinity checks.

Reply:
I've created a new ticket here: https://github.com/peterstace/simplefeatures/issues/402